tests/tool_excimer_script_metadata_test.php

```diff
@@ -188,6 +188,48 @@ public function sampling_limit_provider(): array {
         ];
     }
 
+    /**
+     * Tests script_metadata::get_normalised_relative_script_path().
+     * @dataProvider normalised_relative_script_path_provider
+     * @covers \tool_excimer\script_metadata::get_normalised_relative_script_path
+     * @param string $pathinfo
+     * @param string $expected
+     */
+    public function test_get_normalised_relative_script_path(string $pathinfo, string $expected) {
+        script_metadata::init();
+        $this->assertEquals($expected, script_metadata::get_normalised_relative_script_path($pathinfo));
+    }
+
+    /**
+     * Provider for test_get_normalised_relative_script_path().
+     * @return \string[][]
+     */
+    public function normalised_relative_script_path_provider(): array {
```

Comment:
this is a great start :)

it's probably worth adding this unit test as a first logic commit so you can run it on the old function and then compare to the new function to ensure that nothing has changed unintentionally after you redo the logic of get_normalised_relative_script_path

There are a couple things which I'm pretty sure have changed and should not, like the original function did an ltrim on /, so '/home' should result in 'home' and the only time we want '/' at the start was if the string was empty. So we should explicitly see those changed expectations in the second commits dataprovider changes

Reply:
The leading slash was new logic to make it more consistent with $SCRIPT and the empty case but we can definitely change it back.

I will add the old function back in and compare unit tests.